core/new-gui/src/app/workspace/component/operator-panel/operator-label/operator-label.component.ts

```diff
@@ -16,21 +17,51 @@ import { OperatorSchema } from '../../../types/operator-schema.interface';
 })
 export class OperatorLabelComponent implements AfterViewInit {
 
+  @ViewChild('t') t: any; // t is the specific tooltip window for an operator
   @Input() operator?: OperatorSchema;
   public operatorLabelID: string;
+  private timer: any; // needed to add a delay to tooltip
```

Comment:
@DaiNiaoZhou 
1. type `any` is **strictly prohibited** in our codebase. You need to write down its type, or initialize the object at the same time so TS can infer its type.

2. You current implementation is correct, however, making it more RxJS-ish can help you understand more about RxJS. Here's a better version (but harder to understand) 

```
private mouseEnterObservable = new Subject<void>();
private mouseLeaveObservable = new Subject<void>();

// Angular fills in the variable
ngAfterViewInit() {
    // add these after existing code

    // each mouse enter emission event is converted to an observable, which delays 1500ms, and then emit only if mouseLeave event hasn't emitted during this period
    this.mouseEnterSubject.flatMap(v =>
      of(v).delay(500).takeUntil(this.mouseLeaveSubject)
    ).subscribe(v => {
      if (this.tooltip) {
        this.tooltip.open();
      }
    });

    this.mouseLeaveSubject.subscribe(v => {
      if (this.tooltip) {
        this.tooltip.close();
      }
    });
}

mouseOver(): void {
    this.mouseEnterObservable.next();
} 
mouseLeave(): void {
    this.mouseLeaveObservable.next();
}
```